lib/fetchers/url.rb

```diff
@@ -91,15 +91,25 @@ def self.transform(target)
 
     attr_reader :files, :archive_path
 
-    def initialize(url, opts)
-      @target = url
+    def initialize(target, opts)
+      if target.is_a?(Hash) && target.key?(:url)
```

Comment:
I think you can shorten the if statements to the below if you want (this will also handle if target isn't a String or Hash):
```
@target = target.is_a?(Hash) ? target[:url] : target 
@upstream_sha256 = target.is_a(Hash) ? target[:sha256] : nil
```